storage.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 Teserakt AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e4
+
+import (
+	"bytes"
+	"errors"
+	"io"
+)
+
+// Store defines an interface for E4 storage implementation
+// it is compatible with io.ReadWriteSeeker but redefined to allow this type to be exported
+// from bindings generation
+type Store interface {
+	Write(p []byte) (n int, err error)
+	Read(b []byte) (n int, err error)
+	Seek(offset int64, whence int) (idx int64, err error)
+}
+
+type memoryStore struct {
+	buf   *bytes.Buffer
+	index int64
+}
+
+var _ Store = (*memoryStore)(nil)
+
+// NewMemoryStore creates a new Store in memory
+func NewMemoryStore(buf []byte) Store {
+	return &memoryStore{
+		buf: bytes.NewBuffer(buf),
+	}
+}
+
+func (s *memoryStore) Write(p []byte) (n int, err error) {
+	if s.index < 0 {
+		return 0, io.EOF
+	}
+
+	idx := int(s.index)
```

Comment:
Overflow risk if this hits ~4.3GB, why are we worrying about this condition and also below?
To make circular buffers?